frontend/test/account-send.test.ts

```diff
@@ -0,0 +1,309 @@
+      encryptionCount: 3,
+      expectedCiphertext: '0xd45ffb2b6d4b4ad3bc0682b111cdbceecac938df0c72a0b0a79f7b6be6cfeb3e',
+    },
+  ].forEach(({ advancedMode, usePublicKeyChecked, encryptionCount, expectedCiphertext }) => {
+    it('Correctly encrypts data', () => {
+      const data = encryptAccountData(
+        { recipientAddress, advancedMode, usePublicKeyChecked, pubKey: pubKey },
+        { encryptionCount, viewingPrivateKey: viewingPrivateKey }
+      );
+      expect(data.length).toBe(66); // 32 bytes + 0x
+      expect(data).toBe(expectedCiphertext);
+    });
+  });
+
+  [
+    {
+      ciphertext: '0x9f3873e440b439d4e7561e70b5db28af7abb67257f6353e8d6e057bd2ed16621',
+      encryptionCount: 0,
+      expectedDecryptedData: {
+        advancedMode: true,
+        usePublicKeyChecked: true,
+        address: recipientAddress,
+        pubKey: partialPubKey,
+      },
+    },
+    {
+      ciphertext: '0xac7a4c827e636bb25fbf1ae604c48f0c035c34f526456dc6264b5ee43e3119f2',
+      encryptionCount: 1,
+      expectedDecryptedData: {
+        advancedMode: false,
+        usePublicKeyChecked: false,
+        address: recipientAddress,
+        pubKey: partialPubKey,
+      },
+    },
+    {
+      ciphertext: '0xb609426a8909759990b22756b7f2ce4f8d5ac4685d6e2c40daa830d059950504',
+      encryptionCount: 2,
+      expectedDecryptedData: {
+        advancedMode: true,
+        usePublicKeyChecked: false,
+        address: recipientAddress,
+        pubKey: partialPubKey,
+      },
+    },
+    {
+      ciphertext: '0xd45ffb2b6d4b4ad3bc0682b111cdbceecac938df0c72a0b0a79f7b6be6cfeb3e',
+      encryptionCount: 3,
+      expectedDecryptedData: {
+        advancedMode: false,
+        usePublicKeyChecked: true,
+        address: recipientAddress,
+        pubKey: partialPubKey,
+      },
+    },
+  ].forEach(({ ciphertext, encryptionCount, expectedDecryptedData }) => {
+    it('Data decrypted correctly all true', () => {
+      const data = decryptData(ciphertext, { encryptionCount, viewingPrivateKey: viewingPrivateKey });
+      expect(data).toEqual(expectedDecryptedData); // Recursively checks all properties.
+    });
+  });
+});
+
+describe('encrypt/decrypt relationship', () => {
+  it('Encrypt and decrypt are inverses', () => {
+    // Generate random data and do multiple runs.
+    for (let i = 0; i < 100; i++) {
+      const { recipientAddress, pubKey, viewingPrivateKey, advancedMode, usePublicKeyChecked, encryptionCount } =
+        randomData();
+      const ciphertext = encryptAccountData(
+        { recipientAddress, advancedMode, pubKey, usePublicKeyChecked },
+        { encryptionCount, viewingPrivateKey }
+      );
+      const decryptedData = decryptData(ciphertext, { encryptionCount, viewingPrivateKey });
+      expect(decryptedData).toEqual({
+        advancedMode,
+        usePublicKeyChecked,
+        address: recipientAddress,
+        pubKey: `0x99${pubKey.slice(4, 4 + 22)}`,
+      });
+    }
+  });
+
+  it('bug', () => {
```

Comment:
Took a look, and it looks like we need to do a hexZeroPad in the decrypt method which we are doing in the umbra-js library. It is probably safer to add it in the encrypt method as well.

Reply:
Added zero padding and fixed in 8dc7940